src/main/java/teammates/ui/webapi/action/Action.java

```diff
@@ -97,6 +97,7 @@ public void checkAccessControl() {
     private void initAuthInfo() {
         if (Config.BACKDOOR_KEY.equals(req.getHeader("Backdoor-Key"))) {
             authType = AuthType.ALL_ACCESS;
+            userInfo = new UserInfo(getRequestParamValue(Const.ParamsNames.USER_ID));
```

Comment:
Why was this added?

Reply:
Similar to above comment, it is because archived status is saved under InstructorAttributes and in order to get the archived courses on the instructor, the request must have the instructors user id. Without doing this the user field is null when using Backdoor.